src/common/transformations/src/transformations/common_optimizations/optimize_strided_slice.cpp

```diff
@@ -260,6 +261,134 @@ bool ov::pass::GroupedStridedSliceOptimizer::run_on_model(const std::shared_ptr<
     return graph_rewritten;
 }
 
+struct SliceAttrs {
+    int64_t start, stop, axis;
+};
+
+struct SliceWithAttrs {
+    std::shared_ptr<op::v8::Slice> slice;
+    SliceAttrs attrs;
+};
+
+bool slice_is_suitable_for_optimization(const std::shared_ptr<ov::op::v8::Slice>& op, SliceAttrs& attrs) {
+    const auto& data_rank = op->get_input_partial_shape(0).rank();
+    if (op->get_input_size() != 5 || data_rank.is_dynamic())
+        return false;
+
+    for (size_t i = 1; i < 5; ++i) {
+        auto input_as_constant = ov::as_type_ptr<ov::op::v0::Constant>(op->get_input_node_shared_ptr(i));
+        if (!input_as_constant)
+            return false;
+        if (shape_size(input_as_constant->get_shape()) != 1)
+            return false;
+
+        int64_t value = input_as_constant->cast_vector<int64_t>()[0];
+
+        if (((i == 1 || i == 2) && value < 0) || (i == 3 && value != 1))
```

Comment:
minor: 
we can use unnamed enums here
enum { BEGIN = 1, END, STEP, AXIS};
if ( i == BEGIN || i == END...